### ISSUE TITLE
Renaming a table breaks views using that table

### ISSUE THREAD

**Issue by 0x4261756D** [dolthub/dolt#6691]
**Steps to reproduce:**
```sql
CREATE TABLE a(id int, primary key(id));
CREATE VIEW b AS SELECT id AS c FROM a;
RENAME TABLE a TO d;
SELECT * FROM b;
```
Produces the following error:
```
error on line 1 for query SELECT * FROM b: table not found: a
```

**Comment by fulghum** [comment]
Thanks for reporting this one and providing easy repro steps @0x4261756D. It looks like MySQL will also fail this query, but does give a slightly better error message:
```sql
mysql> CREATE TABLE a(id int, primary key(id));
Query OK, 0 rows affected (0.00 sec)

mysql> CREATE VIEW b AS SELECT id AS c FROM a;
Query OK, 0 rows affected (0.00 sec)

mysql> RENAME TABLE a TO d;
Query OK, 0 rows affected (0.01 sec)

mysql> SELECT * FROM b;
ERROR 1356 (HY000): View 'mydb.b' references invalid table(s) or column(s) or function(s) or definer/invoker of view lack rights to use them
```

Matching MySQL here by updating our error message seems like the right change. That way we match MySQL more exactly and the error message will be more helpful to people so they know the query issue is in the view definition.

**Comment by fulghum** [comment]
@0x4261756D – thanks for reporting this! We've fixed the error message to match MySQL's error message and released that fix in Dolt 1.17.0. Let us know if you find any other issues and we'll be happy to investigate.
